all_test.go

```diff
@@ -0,0 +1,240 @@
+package gitdb_test
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/fobilow/gitdb"
+)
+
+var testDb *gitdb.Connection
+var messageId int
+
+const testData = "/tmp/gitdb-test"
+const dbPath = testData + "/data"
+const fakeRemote = testData + "/online"
+
+//Test flags for more interactivity
+var flagLogLevel int
+var flagFakeRemote bool
+
+func TestMain(m *testing.M) {
+	flag.IntVar(&flagLogLevel, "loglevel", int(gitdb.LOGLEVEL_TEST), "control verbosity of test logs")
+	flag.BoolVar(&flagFakeRemote, "fakerepo", true, "create fake remote repo for tests")
+	flag.Parse()
+
+	gitdb.SetLogLevel(gitdb.LogLevel(flagLogLevel))
+	m.Run()
+}
+
+func setup(t testing.TB) func(t testing.TB) {
+	//fail test if git is not installed
+	if _, err := exec.LookPath("git"); err != nil {
+		t.Error("git is required to run tests")
+	}
+
+	if flagFakeRemote {
+		fakeOnlineRepo(t)
+	}
+
+	testDb = getDbConn(t)
+	messageId = 0
+
+	return teardown
+}
+
+func teardown(t testing.TB) {
+	testDb.Close()
+
+	fmt.Println("truncating test data")
+	cmd := exec.Command("rm", "-Rf", testData)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Errorf("cleanup failed - %s", err.Error())
+	}
+}
+
+func fakeOnlineRepo(t testing.TB) {
+	fmt.Println("creating fake online repo")
+	cmd := exec.Command("mkdir", "-p", fakeRemote)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Errorf("fake repo failed: %s", err.Error())
+		return
+	}
+
+	cmd = exec.Command("git", "-C", fakeRemote, "init", "--bare")
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		t.Errorf("fake repo failed: %s", err.Error())
+		return
+	}
+}
+
+func getDbConn(t testing.TB) *gitdb.Connection {
+	conn, err := gitdb.Open(getConfig())
+	if err != nil {
+		t.Errorf("getDbConn failed: %s", err)
+	}
+	fmt.Println("test db connection opened")
```

Comment:
maybe a  debug log?

Reply:
these are quite useful for now so will keep them and revisit in the future thanks